api/v1alpha1/clusterexternaldns_types.go

```diff
@@ -49,10 +49,10 @@ type ClusterExternalDNSSpec struct {
 	ResourceName string `json:"resourceName"`
 
 	// TenantID is the ID of the Azure tenant where the DNS zones are located.
-	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Optional
 	// +kubebuilder:validation:Format:=uuid
 	// +kubebuilder:validation:Pattern=`[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}`
-	TenantID string `json:"tenantID"`
+	TenantID *string `json:"tenantId,omitempty"`
 
```

Comment:
why change from tenantID to tenantId? would it not break backward compatibility wrt json serialization?

Reply:
Normally yes, but fortunately the CRD fortunately isn't in use anywhere yet. As for the case change - this casing lines up better with OpenAPI conventions